examples/api/lib/cupertino/sheet/cupertino_sheet.2.dart

```diff
@@ -0,0 +1,297 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/cupertino.dart';
+
+/// Flutter code sample for [CupertinoSheetRoute] with restorable state and nested navigation.
+
+void main() => runApp(const RestorableSheetExampleApp());
+
+class RestorableSheetExampleApp extends StatelessWidget {
+  const RestorableSheetExampleApp({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return const CupertinoApp(
+      restorationScopeId: 'sheet-app',
+      title: 'Restorable Sheet',
+      home: RestorableSheet(restorationId: 'sheet'),
+    );
+  }
+}
+
+class RestorableSheet extends StatefulWidget {
+  const RestorableSheet({super.key, this.restorationId});
+
+  final String? restorationId;
+
+  @override
+  State<RestorableSheet> createState() => _RestorableSheetState();
+}
+
+class _RestorableSheetState extends State<RestorableSheet>
+    with RestorationMixin {
+
+  final RestorableInt _counter = RestorableInt(0);
+  late RestorableRouteFuture<int?> _restorableSheetRouteFuture;
+
+  @override
+  void initState() {
+    super.initState();
+    _restorableSheetRouteFuture = RestorableRouteFuture<int?>(
+      onComplete: _changeCounter,
+      onPresent: (NavigatorState navigator, Object? arguments) {
+        return navigator.restorablePush(
+          _counterSheetBuilder,
+          arguments: _counter.value
+        );
+      },
+    );
+  }
+
+  @override
+  String? get restorationId => widget.restorationId;
+
+  @override
+  void restoreState(RestorationBucket? oldBucket, bool initialRestore) {
+    registerForRestoration(_counter, 'count');
+    registerForRestoration(_restorableSheetRouteFuture, 'sheet_route_future');
+  }
+
+  @override
+  void dispose() {
+    _counter.dispose();
+    super.dispose();
+  }
+
+  @pragma('vm:entry-point')
+  static Route<void> _counterSheetBuilder(BuildContext context, Object? arguments) {
+    return CupertinoSheetRoute<int?>(
+      builder: (BuildContext context)  {
+        return Navigator(
+          restorationScopeId: 'nested-nav',
```

Comment:
Can you double check that this works for restoring the current nested page? On Android you could set "Don't keep activities" to true in the device's developer settings, or maybe iOS has something similar. Then navigate to a non-root route in this nested Navigator, switch to another app, switch back, and you should still be at the same nested route.

Reply:
I tested against iOS on a device, and was able to navigate away, stop the app in XCode, then navigate back and be at the correct route and have the right counter value.